open-api.yaml

```diff
@@ -0,0 +1,1614 @@
+    matchesInfo:
+      type: object
+      properties:
+        start:
+          type: integer
+        length:
+          type: integer
+      additionalProperties: false
+      required:
+        - start
+        - length
+      description: ''
+    hit:
+      type: object
+      additionalProperties: true
+      example:
+        id: 25684
+        title: American Ninja 5
+        poster: 'https://image.tmdb.org/t/p/w1280/iuAQVI4mvjI83wnirpD8GVNRVuY.jpg'
+        overview: 'When a scientists daughter is kidnapped, American Ninja, attempts to find her, but this time he teams up with a youngster he has trained in the ways of the ninja.'
+        release_date: '1993-01-01'
+        _formatted:
+          id: 25684
+          title: American Ninja 5
+          poster: 'https://image.tmdb.org/t/p/w1280/iuAQVI4mvjI83wnirpD8GVNRVuY.jpg'
+          overview: 'When a scientists daughter is kidnapped, American <em>Ninja</em>, attempts to find her, but this time he teams up with a youngster he has trained in the ways of the <em>ninja</em>.'
+          release_date: '1993-01-01'
+        _matchesInfo:
+          overview:
+            - start: 49
+              length: 5
+            - start: 155
+              length: 5
+      examples:
+        - id: 25684
+          title: American Ninja 5
+          poster: 'https://image.tmdb.org/t/p/w1280/iuAQVI4mvjI83wnirpD8GVNRVuY.jpg'
+          overview: 'When a scientists daughter is kidnapped, American Ninja, attempts to find her, but this time he teams up with a youngster he has trained in the ways of the ninja.'
+          release_date: '1993-01-01'
+          _formatted:
+            id: 25684
+            title: American Ninja 5
+            poster: 'https://image.tmdb.org/t/p/w1280/iuAQVI4mvjI83wnirpD8GVNRVuY.jpg'
+            overview: 'When a scientists daughter is kidnapped, American <em>Ninja</em>, attempts to find her, but this time he teams up with a youngster he has trained in the ways of the <em>ninja</em>.'
+            release_date: '1993-01-01'
+          _matchesInfo:
+            overview:
+              - start: 49
+                length: 5
+              - start: 155
+                length: 5
+      properties:
+        _formatted:
+          type: object
+          description: Only present if attributesToHighlight is not empty. Copy of normal search result fields with every matching string sequence in the given attribute's field wrapped around an <em> tag.
+          additionalProperties: true
+        _matchesInfo:
+          type: object
+          description: Only present if matches=true. Array of all search query occurrences in all fields.
+          properties:
+            '':
+              $ref: '#/components/schemas/matchesInfo'
+        attribute:
+          type:
+            - string
+            - number
+    documentId:
+      oneOf:
+        - type: number
+        - type: string
+      example: 468219
+    filter:
+      type: array
```

Comment:
can also be string: 

example of valid filter:

nested array:
```json
["something > 1", "genre=comedy", ["genre=horror", "title=batman"]]
```
only string: 
```json
"something > 1 AND genre=comedy AND (genre=horror OR title=batman)"
```

of course both are also mixable
```json
["something > 1 AND genre=comedy", "genre=horror OR title=batman"]
```